packages/SwingSet/src/kernel/vatAdmin/vatAdminWrapper.js

```diff
@@ -21,37 +19,39 @@ export default function setup(syscall, state, helpers) {
   }
 
   function build(E, D) {
-    const vatIdsToRoots = new Map();
+    const vatIdsToResolvers = new Map();
 
     function createVatAdminService(vatAdminNode) {
       return harden({
         createVat(code) {
-          const vatId = D(vatAdminNode).create(code);
-          // If vat creation fails, we get the error message here, (Guaranteed
-          // not to start with a 'v'), so we don't create the promise or node.
-          if (!vatId.startsWith('v')) {
-            return vatId;
+          const result = D(vatAdminNode).create(code);
+          if (result.error) {
+            throw Error(`Vat Creation Error: ${result.error}`);
+          } else {
+            const vatPromise = makePromiseForVat();
+            const vatId = result.ok;
```

Comment:
also cosmetic: maybe we should use `result.vatId` instead of `result.ok`. So the convention would be: if `result.error` exists, there's an error and that property contains the details, else there was no error and other properties (specifically `.vatId`) contains the success results.